libvast/src/system/catalog.cpp

```diff
@@ -405,8 +403,17 @@ catalog(catalog_actor::stateful_pointer<catalog_state> self,
     },
     [=](atom::candidates, vast::uuid lookup_id,
         const vast::expression& expr) -> caf::result<catalog_result> {
+      return self->delegate(static_cast<catalog_actor>(self),
+                            atom::candidates_v, lookup_id, expr,
+                            defaults::latest_partition_version);
```

Comment:
That may actually give missing results if you run VAST on a db created by a newer version. Not sure if we want to support that, but we can avoid the issue by passing an explicit `optional<uint64_>` as max version.

Reply:
I added a comment for this. I don't think it's an actual issue in practice: If a user downgrades from a version with a changed latest partition version we're unlikely to be able to read the newer partitions anyways.

Reply:
Actually, the fix is rather easy: We can just pass the max value.